### ISSUE TITLE
`allow_attributes` lint doesn't detect attributes with whitespace inside brackets

### ISSUE THREAD

**Issue by jamespharaoh** [rust-lang/rust-clippy#16491]
## Summary

The `allow_attributes` lint fails to detect `#[allow(...)]` attributes when there is whitespace inside the brackets, e.g. `#[ allow (dead_code) ]`.

The attribute is parsed correctly by rustc and suppresses warnings as expected, but clippy's lint doesn't recognize it.

## Reproducer

https://github.com/jamespharaoh/clippy-whitespace-issue

```rust
// Spaced allow - NOT detected by lint
#[ allow (dead_code) ]
fn unused_func() {}

// Without spaces - detected correctly
#[allow(dead_code)]
fn another_unused_func() {}

fn main() {}
```

With `Cargo.toml`:
```toml
[package]
name = "lint_test"
version = "0.1.0"
edition = "2024"

[lints.clippy]
allow_attributes = "warn"
```

Running `cargo clippy`:

```
warning: #[allow] attribute found
 --> src/main.rs:6:3
  |
6 | #[allow(dead_code)]
  |   ^^^^^ help: replace it with: `expect`
```

Only the non-spaced version triggers the warning. The spaced version `#[ allow (dead_code) ]` produces no warning, even though:
- The attribute is semantically identical
- It works correctly (the `dead_code` warning is suppressed)
- Rust parses it without issue

## Version

```
rustc 1.93.0 (254b59607 2026-01-19)
binary: rustc
commit-hash: 254b59607d4417e9dffbc307138ae5c86280fe4c
commit-date: 2026-01-19
host: x86_64-unknown-linux-gnu
release: 1.93.0
LLVM version: 21.1.8
```

## Additional context

Looking at the lint implementation in `clippy_lints/src/attrs/mod.rs`:

```rust
if let Some(items) = &attr.meta_item_list()
    && let Some(name) = attr.name()
{
    if matches!(name, sym::allow) && self.msrv.meets(msrvs::LINT_REASONS_STABILIZATION) {
        allow_attributes::check(cx, attr);
    }
```

The condition `attr.meta_item_list()` or `attr.name()` appears to return `None` when there's whitespace between `allow` and `(...)`, even though the attribute is valid.

This likely also affects `allow_attributes_without_reason` since it uses similar logic.

**Comment by vishnupoddar12** [comment]
I found that attr.meta_item_list() or attr.name() are not returning None. attr.meta_item_list() actually parses attributes with internal whitespace (like #[ allow (unused) ]) perfectly fine. 

The reason the lint fails to trigger is due to a false positive in is_from_proc_macro. clippy_utils::is_from_proc_macro works by comparing the source code text against a rigid expected pattern. If the source code text matches the pattern, it's considered user-written. If it doesn't match, it's assumed to be macro-generated.

This specific code in clippy_utils/src/check_proc_macro.rs is the culprit:
```rust
fn attr_search_pat(attr: &Attribute) -> (Pat, Pat) {
    match attr.kind {
        AttrKind::Normal(..) => {
            if let Some(name) = attr.name() {
                // NOTE: This will likely have false positives, like `allow = 1`
                let ident_string = name.to_string();
                if attr.style == AttrStyle::Outer {
                    (
                        Pat::OwnedMultiStr(vec!["#[".to_owned() + &ident_string, ident_string]),
                        Pat::Str(""),
                    )
                } else {
                    (
                        Pat::OwnedMultiStr(vec!["#![".to_owned() + &ident_string, ident_string]),
                        Pat::Str(""),
                    )
                }
            } else {
                (Pat::Str("#"), Pat::Str("]"))
            }
        },
```

**Comment by profetia** [comment]
@vishnupoddar12 Hi! I forgot to claim this issue when starting working on it and I already implemented a fix. Do you mind transferring it to me?

**Comment by vishnupoddar12** [comment]
Hi @profetia Typically, I would be happy to step aside, but this would actually be my very first contribution to the Clippy community, so I am really excited about getting it across the finish line. Like you I have also implemented the fix(using a new **Pat::Attr** variant in **check_proc_macro** to handle whitespace structurally) and verified it.

Can I send the PR and perhaps you could review it to ensure I haven't missed anything? Your experience would be really helpful to guide me through my first PR!

**Comment by profetia** [comment]
I see. You can open a PR and I will help review it. Also, feel free to cherry pick from my implementation (https://github.com/profetia/rust-clippy/commit/484120727a3f586ad2086dfc2e30d5ad20da77dd)
